daemon/algod/api/server/common/handlers.go

```diff
@@ -48,6 +50,60 @@ func GenesisJSON(ctx lib.ReqContext, context echo.Context) {
 	w.Write([]byte(lib.GenesisJSONText))
 }
 
+// Consensus returns protocol consensun information
+func Consensus(ctx lib.ReqContext, context echo.Context) {
+	// swagger:operation GET /consensus Consensus
+	//---
+	//     Summary: Returns the protocol consensus information.
+	//     Produces:
+	//     - application/json
+	//     Schemes:
+	//     - http
+	//     Parameters:
+	//       - name: version
+	//         in: query
+	//         type: string
+	//         description: Return information about a specific version, if exists.
+	//     Responses:
+	//       200:
+	//         description: The protocol consensus details in json
+	//         schema: {type: string}
+	//       404:
+	//         description: Version Not Found
+	//         schema: {type: string}
+	//       default: { description: Unknown Error }
+	w := context.Response().Writer
+
+	// Check for unknown query parameters.
+	validQueryParams := map[string]bool{
+		"version": true,
+	}
+	for name := range context.QueryParams() {
+		if _, ok := validQueryParams[name]; !ok {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte(fmt.Sprintf("Unknown parameter detected: %s", name)))
+			return
+		}
+	}
+
+	version := context.QueryParam("version")
+	if len(version) == 0 {
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		w.Write(protocol.EncodeJSON(config.Consensus))
```

Comment:
maybe return not everything but keys (consensus versions) ?

Reply:
May be, but also may be it won't be necessary to expose the whole protocol information but only useful stuff.